### PR TITLE
chore: upgrade to Sentry 8

### PR DESCRIPTION
**Pull request by matthewwalsh0**
## **Description**

Upgrade the Sentry packages from `7.53.0` to `8.19.0`.

Specifically:

- Refactor custom `FilterEvents` integration into object returned by function.
- Remove usages of `hub`.
- Fix notification errors when changing metrics preference.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25999?quickstart=1)

## **Manual testing steps**

Full Sentry regression.

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by matthewwalsh0** [comment]
@SocketSecurity ignore-all

**Comment by matthewwalsh0** [comment]
@metamaskbot update-policies

**Comment by DDDDDanica** [comment]
Tested locally, work as expected

File: yarn.lock [hunk 1]
```diff
@@ -7922,29 +7922,64 @@ __metadata:
...
+  dependencies:
+    "@sentry-internal/replay": "npm:8.19.0"
+    "@sentry/core": "npm:8.19.0"
+    "@sentry/types": "npm:8.19.0"
+    "@sentry/utils": "npm:8.19.0"
+  checksum: 10/1f379c141884b448c56fcd663b8acc0ff1c12d50a2b9db37f9552eb2bc8c99a970114f80e58c8c4fcd61f933f9a15f58dc6cbe6f4297bb574d6772be8f41c5bf
+  languageName: node
+  linkType: hard
+
+"@sentry-internal/replay@npm:8.19.0":
```
[the remark this answers is not in the document]
Comment (matthewwalsh0) [reply yarn.lock:7959]: Is it standard to do this as we go?

This changes loads of attribution versions so I wasn't sure if this is typically done in a dedicated PR prior to release?

**Review by OGPoyraz — DISMISSED** [review]

**Review by vinistevam — DISMISSED** [review]

File: yarn.lock (hunk 1 above)
[the remark this answers is not in the document]
Comment (DDDDDanica) [reply yarn.lock:7959]: i'm not 100% sure about this one. we just doing research of legacy package and this came out, we can target in another pr to unblock

**Review by DDDDDanica — APPROVED** [review]

**Review by Prithpal-Sooriya — APPROVED** [review]

File: ui/hooks/metamask-notifications/useMetametrics.ts [hunk 2]
```diff
@@ -81,7 +82,7 @@ export function useDisableMetametrics(): {
     setError(null);
 
     try {
-      if (!isProfileSyncingEnabled) {
+      if (isProfileSyncingEnabled) {
```
[the remark this answers is not in the document]
Comment (matthewwalsh0) [reply ui/hooks/metamask-notifications/useMetametrics.ts:85]: Also reproduced by @OGPoyraz.

**Pull request merged**